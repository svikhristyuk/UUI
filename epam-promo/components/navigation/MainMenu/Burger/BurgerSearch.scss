@use  '../../../../assets/styles/index' as *;
@import '../../../../assets/styles/colorvars/navigation/mainMenu/burger/burgerSearch-colorvars';

.search-input {
    @include burger-search-color();
    height: 60px;
    background-color: var(--background-color);
    fill: var(--fill);
    padding: 0 24px;
    flex-shrink: 0;
    width: auto;

    :global(.uui-input) {
        color: var(--color);
        font-family: $font-sans;

        &::placeholder {
            color: var(--placeholder-color);
        }
    }

    :global(.uui-caption),
    :global(.uui-input) {
        padding: 18px 16px;
        line-height: 24px;
        font-size: 14px;
    }

    :global(.uui-icon) {
        svg {
            height: 16px;
            max-width: 16px;
        }
    }

    :global(.uui-icon-cancel) {
        svg {
            height: 24px;
            max-width: 24px;
        }

        &:hover {
            cursor: pointer;
        }
    }
}
