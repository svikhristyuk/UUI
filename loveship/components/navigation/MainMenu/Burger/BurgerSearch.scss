@use '../../../../assets/styles/scss/index'  as *;

.search-input {
    height: 60px;
    background-color: $night800;
    fill: $night500;
    padding: 0 24px;
    flex-shrink: 0;
    width: auto;

    :global(.uui-input) {
        color: $night300;
        font-family: $font-sans;

        &::placeholder {
            color: $night500;
        }
    }

    :global(.uui-caption),
    :global(.uui-input) {
        padding: 18px 16px;
        line-height: 24px;
        font-size: 14px;
    }

    :global(.uui-icon) {
        svg {
            height: 16px;
            max-width: 16px;
        }
    }

    :global(.uui-icon-cancel) {
        svg {
            height: 24px;
            max-width: 24px;
        }

        &:hover {
            cursor: pointer;
        }
    }
}
