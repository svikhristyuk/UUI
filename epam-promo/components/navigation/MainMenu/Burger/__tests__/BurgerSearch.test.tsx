import React from 'react';
import { BurgerSearch } from '../BurgerSearch';
import renderer from 'react-test-renderer';
import { ReactComponent as CalendarIcon } from '../../../../../icons/calendar-18.svg';

describe('BurgerSearch', () => {
    it('should be rendered correctly', () => {
        const tree = renderer
            .create(<BurgerSearch
                value={ null }
                onValueChange={ () => {} }
            />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should be rendered correctly', () => {
        const tree = renderer
            .create(<BurgerSearch
                value={ null }
                onValueChange={ () => {} }
                onAccept={ () => {} }
                onCancel={ () => {} }
                icon={ CalendarIcon }
                iconPosition='right'
                isDropdown
                isOpen
            />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});


