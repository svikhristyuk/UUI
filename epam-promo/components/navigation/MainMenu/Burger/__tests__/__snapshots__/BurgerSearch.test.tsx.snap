// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BurgerSearch should be rendered correctly 1`] = `
<div
  className="container uui-input-box -clickable searchInput"
  onBlur={[Function]}
  onFocus={[Function]}
  tabIndex={-1}
>
  <div
    className="container uui-icon uui-enabled"
    style={Object {}}
  >
    <SvgrURL
      className=""
    />
  </div>
  <input
    autoFocus={true}
    className="uui-input"
    onChange={[Function]}
    onKeyDown={[Function]}
    type="text"
    value=""
  />
</div>
`;

exports[`BurgerSearch should be rendered correctly 2`] = `
<div
  className="container uui-input-box -clickable searchInput"
  onBlur={[Function]}
  onFocus={[Function]}
  tabIndex={-1}
>
  <div
    className="container uui-icon uui-enabled"
    style={Object {}}
  >
    <SvgrURL
      className=""
    />
  </div>
  <input
    autoFocus={true}
    className="uui-input"
    onChange={[Function]}
    onKeyDown={[Function]}
    type="text"
    value=""
  />
</div>
`;
